addons/account/models/account_move.py

```diff
@@ -3457,6 +3457,25 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
 
         return reverse_moves
 
+    def _need_reverse(self):
+        self.ensure_one()
+        lock_date = self.company_id._get_user_fiscal_lock_date()
+        return self.inalterable_hash or self.date <= lock_date
+
+    def _unlink_or_reverse(self):
+        if not self:
+            return
+        to_reverse = self.env['account.move']
+        to_unlink = self.env['account.move']
+        for move in self:
+            if move._need_reverse():
+                to_reverse += move
+            else:
+                to_unlink += move
```

Comment:
```suggestion
            if move._can_be_unlinked():
                to_unlink += move
            else:
                to_reverse += move
```

I would prefer this version because it's not because it "needs reverse" that we reverse it, but instead we cannot unlink it, so as a fallback we reverse it. I hope that makes sense.

But it's totally detable, let me know what you think :)

Reply:
Fine by me, it makes more sense :+1: